### DISCUSSION TITLE
Next.js _next/data JSON routes are returning HTML

### DISCUSSION THREAD

**Discussion by good-idea** [vercel/vercel#5796] [Help]
I'm having an issue where the JSON page data for an SSG Next.js app has started returning HTML instead of `pageProps` as it should. I've narrowed it down to being some change within my `yarn.lock` file.

The HTML being returned is the page-specific content - everything rendered as a child of `_app.tsx`

**Deployment 1** - working

- data: https://baileyhikawa-ln0ce76h1.vercel.app/_next/data/WmOxswE8KsekFlbWyj1sC/products/origami-mask-collab-with-lefty-in-multi-canvas.json
 - commit: https://github.com/good-idea/hikawa.studio/commit/4c56491
 
 **Deployment 2** - not working
 
 - data: https://baileyhikawa-lgr7b5he0.vercel.app/_next/data/67nB3rSjzJdOUKfcI_8rK/products/origami-mask-collab-with-lefty-in-multi-canvas.json <- Returns HTML!
 - commit: https://github.com/good-idea/hikawa.studio/commit/7c6606c7c3eedaf1478701a4772d5a39b704fb10
 
 The only change here was removing `yarn.lock` then reinstalling.

I'm not sure if this helps, but when i create a build locally and run `NODE_ENV=production yarn next start`, the HTML returned from the bad deployment is the same as `.next/server/pages/products/[productHandle].html`.

**Comment by huda0816** [comment]
Sorry for necoing that post but did you find a solution for your problem. We are facing the same problem right now and I cannot fix it.

**Reply by mattlevine** [reply]
Any luck? I've just run into this as well.

**Reply by amyegan** [reply]
It looks like the original example deployments are deleted now, so I can't see exactly what was going wrong. Can you [start a new discussion](https://github.com/vercel/community/discussions/new) with some example links and a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can take a closer look?

**Reply by mattlevine** [reply]
I believe that this discussion thread is about the same topic and has an example, I'll look to create a more isolated example at the same time.

https://github.com/vercel/next.js/discussions/39505

**Reply by mattlevine** [reply]
Actually someone posted a pretty precise example of it happening on that thread.

https://codesandbox.io/s/e0eu71

**Reply by huda0816** [reply]
This may sound stupid but I was digging in core files for hours and could not find a solution. Then I did something which I should have done in the beginning: I updated nextjs. Which solved my problem.

**Reply by mattlevine** [reply]
Thanks for the update! I thought I tested with 12.3.1 yesterday, but perhaps I went to fast through the steps and missed something.
